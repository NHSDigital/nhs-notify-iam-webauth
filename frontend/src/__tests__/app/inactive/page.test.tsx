import { render } from '@testing-library/react';
import { ReadonlyURLSearchParams, useSearchParams } from 'next/navigation';
import InactivePage from '../../../app/inactive/page';

jest.mock('../../../components/molecules/SignOut/SignOut', () => ({
  SignOut: ({ children }: { children: React.ReactNode }) => children,
}));

jest.mock('next/navigation', () => ({
  // Note: We have to requireActual because we need the concrete implementation of ReadonlyURLSearchParams
  ...jest.requireActual('next/navigation'),
  useSearchParams: jest.fn(),
}));

const useSearchParamsMock = jest.mocked(useSearchParams);

describe('Inactive page', () => {
  beforeEach(() => jest.resetAllMocks);

  test('matches snapshot', async () => {
    const mockSearchParams = new ReadonlyURLSearchParams({
      redirect: '/my-redirect-path',
    });

    useSearchParamsMock.mockReturnValue(mockSearchParams);

    const container = render(<InactivePage />);

    expect(container.asFragment()).toMatchSnapshot();
  });

  test('should have no redirect path', async () => {
    useSearchParamsMock.mockReturnValue(new ReadonlyURLSearchParams());

    const { findByRole } = render(<InactivePage />);

    const signInLink = await findByRole('button', { name: 'Sign in' });

    expect(signInLink.getAttribute('href')).toEqual('/auth');
  });
});
