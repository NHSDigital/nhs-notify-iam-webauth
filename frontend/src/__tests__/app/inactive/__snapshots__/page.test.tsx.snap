// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inactive page matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    You've been signed out
  </h1>
  <p>
    You've been signed out because you've not used this service for 15 minutes
  </p>
  <p>
    Any unsaved changes have been lost.
  </p>
  <p>
    Sign in again to create and submit a template to NHS Notify.
  </p>
  <a
    aria-disabled="false"
    class="nhsuk-button"
    draggable="false"
    href="/auth?redirect=%2Fmy-redirect-path"
    id="inactive-sign-in"
    role="button"
  >
    Sign in
  </a>
</DocumentFragment>
`;
